Fix RRef type annotations

r-barnes opened the pull request:
Test Plan: Sandcastle

Reviewed By: H-Huang

Differential Revision: D47334579

Skylion007 commented on torch/distributed/rpc/__init__.py:
```diff
@@ -81,11 +81,11 @@ def is_available():
     __all__ = __all__ + api.__all__ + backend_registry.__all__  # noqa: PLE0605
```
import Optional type here

H-Huang reviewed (approved):
approved internally

r-barnes committed:
Fix RRef type annotations (#104876)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/104876

Test Plan: Sandcastle

Reviewed By: H-Huang

Differential Revision: D47334579

fbshipit-source-id: 2d93c642b4029a218c249466d0fad49c8049b613
